src/binding/python/Attributable.cpp

```diff
@@ -0,0 +1,173 @@
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, int64_t const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, float const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, double const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, long double const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::string const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<char> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<unsigned char> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<int16_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<int32_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<int64_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<uint16_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<uint32_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<int64_t> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<float> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<double> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<long double> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::vector<std::string> const& v ) {
+            attr.setAttribute(key, v);
+        })
+        .def("set_attribute", []( Attributable & attr, std::string const& key, std::array< double, 7 > const& v ) {
```

Comment:
Will Python ever store ```std::array< double, 7 >```?
I would assume any list with doubles of length 7 would be caught as a ```std::vector<double>``` first.

Reply:
To expand on this a little:
```python
record.set_unit_dimension([...])
```
will probably work, but
```python
udim = [...]
record.set_attribute('unitDimension', udim)
```
will probably not.

Reply:
This is precicely my concern with #255, btw. It is bad enough if we have this problem in Python, having it in out code C++ API will only be worse.

Reply:
Why does it even matter if we pass the unitDimension as `std::vector<double>` or `array< double, 7>`? In the end it must be and is the same representation in the files (intentionally).

I will still try the examples you post and see where it specializes to.

Reply:
> In the end it must be and is the same representation in the files (intentionally).

True, but only for the backends we currently employ. You could think of a pure serialization backend where this does not apply.

I was worried about retrieval, but we are covered:
https://github.com/openPMD/openPMD-api/blob/4832b11a91a70a9a6b7d91e119222fb02602b5ea/include/openPMD/backend/BaseRecord.hpp#L190-L206

Reply:
I think that's fine. We just say for unitDimensions: "it's an array as usual, but of size 7"